staff/web/vhost/redirect_check.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+import argparse
+import socket
+
+import requests as r
+from ocflib.vhost.web import get_vhosts
+
+out = 'redirect.log'
+errors = 'errors_redir.log'
+special_strings = ['asuc', 'ocf']
+
+
+def striphttp(name):
+    return name.split('/')[2]
+
+
+def is_special(url_string):
+    return any([special_string in url_string for special_string in
+                special_strings])
+
+
+def check_vhosting():
+    with open(out, 'w') as log, open(errors, 'w') as error_file:
+        vhosts = get_vhosts()
+        vhost_urls = []
+        for vhost_url in vhosts.keys():
+            if any(is_special(url) for url in {vhost_url} | set(vhosts[vhost_url]['aliases'])):
+                continue
+            vhost_urls.append('https://' + vhost_url)
+        # For log niceness
+        vhost_urls.sort()
+        baseIP = socket.gethostbyname('ocf.berkeley.edu')
+        for site in vhost_urls:
+            try:
+                print('Opening', site)
+                siter = r.get(site, timeout=10)
+                newIP = socket.gethostbyname(striphttp(siter.url))
+                if baseIP != newIP:
+                    print('\t', newIP)
+                    log.writelines(site + '\n')
```

Comment:
`writelines` is for writing a list of strings. If you just want to write a string, use `write` instead. If you use `writelines`, then it will call `write` on each character individually, which is dumb.

Having said that, I would actually write this as `print(site, file=log)` instead.